spokestack/asr/cloud_client.py

```diff
@@ -0,0 +1,206 @@
+"""
+This module contains the websocket logic used to communicate with
+Spokestack's cloud-based ASR service.
+"""
+import base64
+import hashlib
+import hmac
+import json
+from typing import Any, Dict, List
+
+import numpy as np  # type: ignore
+from websocket import WebSocket  # type: ignore
+
+
+class CloudClient:
+    """ Spokestack client for cloud based speech to text
+
+        Args:
+            socket_url (str): url for socket connection
+            key_id (str): identity from spokestack api credentials
+            key_secret (str): secret key from spokestack api credentials
+            audio_format (str): format of input audio
+            sample_rate (int): audio sample rate (kHz)
+            language (str): language for recognition
+            limit (int): Limit of messages per api response
+            idle_timeout (Any): Time before client timeout. Defaults to None
+    """
+
+    def __init__(
+        self,
+        socket_url: str,
+        key_id: str,
+        key_secret: str,
+        audio_format: str = "PCM16LE",
+        sample_rate: int = 16000,
+        language: str = "en",
+        limit: int = 10,
+        idle_timeout: Any = None,
```

Comment:
this should be union of `None` or `float`
for example, strings should not be accepted here